src/Symfony/Component/Mailer/Transports.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Mailer;
+
+use Symfony\Component\Mailer\Exception\InvalidArgumentException;
+use Symfony\Component\Mailer\Exception\LogicException;
+use Symfony\Component\Mailer\Transport\TransportInterface;
+
+/**
+ * @author Fabien Potencier <fabien@symfony.com>
+ */
+class Transports
+{
+    private $transports;
+    private $default;
+
+    /**
+     * @param TransportInterface[] $transports
+     */
+    public function __construct(iterable $transports)
```

Comment:
Couldn't we use sth like: `__construct(Transport ...$transports)`?

Reply:
Look, the keys of the iterator are used. And the check is done on the return type of getTransport. That's way enough for the use case.